lib/ExecutionEngine/ExecutionEngine.cpp

```diff
@@ -114,20 +114,23 @@ void glow::updateInputPlaceholdersByName(PlaceholderBindings &bindings,
 void ExecutionEngine::runInternal(ExecutionContext &context,
                                   llvm::StringRef name) {
   std::unique_ptr<ExecutionContext> contextPtr(&context);
+  std::unique_ptr<ExecutionContext> contextOut;
   std::promise<void> runPromise;
   auto fut = runPromise.get_future();
   Error runErr = Error::empty();
-  hostManager_->runNetwork(
-      name, std::move(contextPtr),
-      [&runPromise, &runErr](runtime::RunIdentifierTy, Error err,
-                             std::unique_ptr<ExecutionContext> contextPtr) {
-        // Don't delete context.
-        contextPtr.release();
-        runErr = std::move(err);
-        runPromise.set_value();
-      });
+  hostManager_->runNetwork(name, std::move(contextPtr),
+                           [&runPromise, &runErr, &contextOut](
+                               runtime::RunIdentifierTy, Error err,
+                               std::unique_ptr<ExecutionContext> contextPtr) {
+                             contextOut = std::move(contextPtr);
+                             runErr = std::move(err);
+                             runPromise.set_value();
+                           });
 
   fut.wait();
+  hostManager_->ensureOutputsAvailable(*contextOut.get());
```

Comment:
It would be nice to be able to control this behaviour somehow so if we're using the EE but we want to leave Tensors of the device we could.